### PR TITLE
Added Detection Rule for Malicious OneNote Commands

### PR DESCRIPTION
**Pull request by arnydo**

### CODE REVIEW

File: detection-rules/attachment_malicious_onenote_commands.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+description: |
+  Scans for OneNote attachments that contain suspicious commands that may indicate malicious activity.
+references:
+  - "https://www.trustedsec.com/blog/new-attacks-old-tricks-how-onenote-malware-is-evolving/"
+type: "rule"
+severity: "medium"
+source: |
+type.inbound
+and any(attachments,(.file_extension == 'one')
+and any(beta.binexplode(.), any(.scan.strings.strings, strings.ilike(., "*WshShell*", "*ExecuteCmdAsync*", "*CreateObject*", "*Wscript.Shell*","*schtasks*", "*CreateProcess*","*winmgmts*", "*SetEnvironmentVariable*"))))
```
Comment (jkamdjou) [inline detection-rules/attachment_malicious_onenote_commands.yml:11]:
```suggestion
    type.inbound
    and any(attachments, .file_extension == 'one'
        and any(beta.binexplode(.),
            any(.scan.strings.strings,
                strings.ilike(., 
                    "*WshShell*",
                    "*ExecuteCmdAsync*",
                    "*CreateObject*",
                    "*Wscript.Shell*",
                    "*schtasks*",
                    "*CreateProcess*",
                    "*winmgmts*",
                    "*SetEnvironmentVariable*",
                )
            )
        )
    )
```

Comment (jkamdjou) [reply detection-rules/attachment_malicious_onenote_commands.yml:11]: added preceding spaces to make it valid YAML, and a tweak to formatting for readability

File: detection-rules/attachment_malicious_onenote_commands.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+tags: 
+  - "Suspicious attachment"
+  - "Malware"
+  - "Office exploit"
```
Comment (jkamdjou) [inline detection-rules/attachment_malicious_onenote_commands.yml:15]:
```suggestion
```

Comment (jkamdjou) [reply detection-rules/attachment_malicious_onenote_commands.yml:15]: i think we'd call this an exploit if there was a vulnerability / potential for a CVE, but in this case it looks like it's a "feature" 😅

File: detection-rules/attachment_malicious_onenote_commands.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: "Attachment: Malicious OneNote Commands"
+description: |
+  Scans for OneNote attachments that contain suspicious commands that may indicate malicious activity.
+references:
+  - "https://www.trustedsec.com/blog/new-attacks-old-tricks-how-onenote-malware-is-evolving/"
+type: "rule"
+severity: "medium"
```
Comment (jkamdjou) [inline detection-rules/attachment_malicious_onenote_commands.yml:7]:
```suggestion
severity: "high"
```

Comment (jkamdjou) [reply detection-rules/attachment_malicious_onenote_commands.yml:7]: if there's no legit use case for these commands, we might make this `high`

**Review by jkamdjou — COMMENTED** [review]
🔥 this is awesome, thanks @arnydo! a couple small suggestions

**Comment by arnydo** [comment]
I appreciate the feedback @jkamdjou!! I accepted your recommendations and will keep them in mind for next time :-)

File: detection-rules/attachment_malicious_onenote_commands.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            any(.scan.strings.strings,
+                strings.ilike(., 
+                    "*WshShell*",
+                    "*ExecuteCmdAsync*",
+                    "*CreateObject*",
+                    "*Wscript.Shell*",
+                    "*schtasks*",
+                    "*CreateProcess*",
+                    "*winmgmts*",
+                    "*SetEnvironmentVariable*",
```
Comment (morriscode) [inline detection-rules/attachment_malicious_onenote_commands.yml:21]:
```suggestion
                    "*CreateProcess*",
                    "*winmgmts*",
                    "*SetEnvironmentVariable*",
                    "*powershell*"
```

@arnydo Thoughts on adding powershell?

Comment (arnydo) [reply detection-rules/attachment_malicious_onenote_commands.yml:21]: Absolutely!

File: detection-rules/attachment_malicious_onenote_commands.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+name: "Attachment: Malicious OneNote Commands"
+description: |
+  Scans for OneNote attachments that contain suspicious commands that may indicate malicious activity.
+references:
+  - "https://www.trustedsec.com/blog/new-attacks-old-tricks-how-onenote-malware-is-evolving/"
+type: "rule"
```
Comment (jkamdjou) [inline detection-rules/attachment_malicious_onenote_commands.yml:6]:
```suggestion
type: "rule"
authors:
  - twitter: "Kyle_Parrish_"
    name: "Kyle Parrish"
```

don't forget to give yourself credit :)

File: detection-rules/attachment_malicious_onenote_commands.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                    "*Wscript.Shell*",
+                    "*schtasks*",
+                    "*CreateProcess*",
+                    "*winmgmts*",
+                    "*SetEnvironmentVariable*",
+                    "*powershell*"
+                )
+            )
+        )
+    )
```
[the remark this answers is not in the document]
Comment (arnydo) [reply detection-rules/attachment_malicious_onenote_commands.yml]: Thank you Sam!!!

**Review by jkamdjou — APPROVED** [review]
🙌 🙌

**Pull request merged**